### PR TITLE
SDP-488: store the Anchor Platform's tx ID when registering a receiver's wallet

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Store the Anchor Platform's tx ID when registering a receiver's wallet

### Why

So we can PATCH the AP with the success/error states later.

### Further discussion

I'm storing the AP ID in the receiver wallet entry, because this is the DB row that gets updated when we complete the SEP24 registration.

We don't know exactly the amount that will be updated there, but we can update the amount when we PATCH the success/failure responses.

Alternatively, we could store the AnchorPlatform ID in a payment row, but since that is not used in the registration flow, it seemed inconsistent to rely on that table for this part of the flow while the registration does not rely on payments.

#### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [x] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/data/receivers_wallet_test.go [hunk 1]
```diff
@@ -554,10 +559,11 @@ func Test_UpdateReceiverWallet(t *testing.T) {
 		require.NoError(t, err)
 
 		assert.Equal(t, RegisteredReceiversWalletStatus, receiverWalletUpdated.Status)
+		assert.Equal(t, "test-anchor-tx-platform-id", receiverWalletUpdated.AnchorPlatformTransactionID)
 		assert.Equal(t, "GBLTXF46JTCGMWFJASQLVXMMA36IPYTDCN4EN73HRXCGDCGYBZM3A444", receiverWalletUpdated.StellarAddress)
 		assert.Equal(t, "123456", receiverWalletUpdated.StellarMemo)
 		assert.Equal(t, "id", receiverWalletUpdated.StellarMemoType)
-		require.NotEmpty(t, receiverWalletUpdated.OTPConfirmedAt)
+		assert.WithinDuration(t, now, *receiverWalletUpdated.OTPConfirmedAt, 100*time.Millisecond)
```
Comment (CaioTeixeira95) [inline internal/data/receivers_wallet_test.go:566]: Interesting!

File: internal/serve/httphandler/verifiy_receiver_registration_handler.go [hunk 2]
```diff
@@ -148,67 +153,86 @@ func (v VerifyReceiverRegistrationHandler) processReceiverVerificationPII(ctx co
...
+	rw.OTPConfirmedAt = &now
+	rw.Status = data.RegisteredReceiversWalletStatus
+	rw.StellarAddress = sep24Claims.SEP10StellarAccount()
+	rw.StellarMemo = sep24Claims.SEP10StellarMemo()
 	if sep24Claims.SEP10StellarMemo() != "" {
-		receiverWallet.StellarMemo = sep24Claims.SEP10StellarMemo()
-		receiverWallet.StellarMemoType = "id"
+		rw.StellarMemoType = "id"
+	} else {
+		rw.StellarMemoType = ""
```
Comment (CaioTeixeira95) [inline internal/serve/httphandler/verifiy_receiver_registration_handler.go:198]: Here you can do:

```go
        rw.StellarMemoType = ""
        if sep24Claims.SEP10StellarMemo() != "" {
		rw.StellarMemoType = "id"
	}
```

File: internal/serve/httphandler/verifiy_receiver_registration_handler.go [hunk 3]
```diff
@@ -253,7 +277,7 @@ func (v VerifyReceiverRegistrationHandler) VerifyReceiverRegistration(w http.Res
 		}
 
 		// STEP 5: PATCH transaction on the AnchorPlatform
```
Comment (CaioTeixeira95) [inline internal/serve/httphandler/verifiy_receiver_registration_handler.go:279]: We can add to this comment that we also update the `AP Transaction ID` on the `receiver_wallets` table.

**Review by CaioTeixeira95 — APPROVED** [review]
LGTM!

File: internal/serve/httphandler/verifiy_receiver_registration_handler.go (hunk 3 above)
Comment (marcelosalloum) [reply internal/serve/httphandler/verifiy_receiver_registration_handler.go]: Addresed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/44/commits/9876661a08d58850949a2c00bfa6d428f627b7d5

File: internal/serve/httphandler/verifiy_receiver_registration_handler.go (hunk 2 above)
Comment (marcelosalloum) [reply internal/serve/httphandler/verifiy_receiver_registration_handler.go]: Addressed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/44/commits/9876661a08d58850949a2c00bfa6d428f627b7d5.

**Pull request merged**